sharktank/tests/models/llama/test_llama.py

```diff
@@ -61,6 +71,57 @@ def testUnsharded(self):
         assert pytest.approx(0.583, 1e-2) == cross_entropy
 
 
+@pytest.mark.expensive
+@pytest.mark.skipif(
+    not torch.cuda.is_available(),
+    reason="This test is too slow on CPU. Could take up to 5 minutes on a 64 core CPU.",
+)
+def test_pruned_llama3_405b_f4_pipeline_parallel_eager_vs_eager_perplexity(
+    deterministic_random_seed, model_artifacts: dict[str, str]
+):
+    """Verify that a pipeline-parallel pruned (removed layers) variant of the 405B f4
+    model produces the same perplexity as a the reference variant that is not
+    pipeline-parallel.
+    We don't care if the perplexity is high. Just that it is the same against the reference."""
+    device = torch.device("cuda")
+    batch_size = 4
+    prune_to_block_count = 3
+    pipeline_parallelism_size = 2
+
+    parameters_path = model_artifacts["llama3_1_405b_instruct_f4_model_path"]
+    dataset = Dataset.load(parameters_path)
+
+    reference_config = LlamaModelConfig.from_dataset(dataset)
+    # TODO: remove when the IRPA has the correct value
```

Comment:
Please move to the pre-shuffled IRPA for this PR.

This PR is still blocked by: https://github.com/iree-org/iree-turbine/pull/1157.